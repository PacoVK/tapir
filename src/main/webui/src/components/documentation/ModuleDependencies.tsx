import React from "react";
import { Divider, Stack, Typography } from "@mui/material";
import { ModuleDependenciesProps } from "../../types";
import Container from "@mui/material/Container";

const ModuleDependencies = (props: ModuleDependenciesProps) => {
  const { modules, providers, ...other } = props;
  return (
    <>
      <Typography variant="h4">Module Dependencies</Typography>
      <Typography sx={{ mt: "1em" }}>Todo hint </Typography>
      <Stack
        justifyContent="center"
        alignItems="left"
        spacing={2}
        sx={{ mt: "2em", mb: "2em" }}
      >
        {modules ? (
          modules.map((module) => {
            return (
              <>
                <Typography>
                  <strong>{module.source}</strong>
                </Typography>
                <Container maxWidth="sm">
                  <Typography>
                    Version:{" "}
                    <em>{module.version ? module.version : "latest"}</em>
                  </Typography>
                </Container>
              </>
            );
          })
        ) : (
          <Typography variant="h6">
            This module has no external module dependencies
          </Typography>
        )}
      </Stack>
      <Divider sx={{ mb: "2em" }} />
      <Typography variant="h4">Provider Dependencies</Typography>

      <Typography sx={{ mt: "1em" }}>Todo hint </Typography>
      <Stack
        justifyContent="center"
        alignItems="left"
        spacing={2}
        sx={{ mt: "15px", mb: "15px" }}
      >
        {providers
          ? providers.map((provider) => {
              return (
                <>
                  <Typography>
                    <strong>{provider.name}</strong>
                  </Typography>
                  <Container maxWidth="sm">
                    {provider.alias ? (
                      <Typography>
                        Alias: <em>{provider.alias}</em>
                      </Typography>
                    ) : null}
                    <Typography>
                      Version:{" "}
                      <em>{provider.version ? provider.version : "latest"}</em>
                    </Typography>
                  </Container>
                </>
              );
            })
          : null}
      </Stack>
    </>
  );
};

export default ModuleDependencies;
