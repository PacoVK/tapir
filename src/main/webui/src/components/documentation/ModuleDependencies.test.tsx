import { render } from "@testing-library/react";
import ModuleDependencies from "./ModuleDependencies";

describe("<ModuleDependencies /> spec", () => {
  it("renders the ModuleDependencies", () => {
    const view = render(
      <ModuleDependencies
        modules={[
          { source: "terraform-aws-modules/security-group/aws", version: "" },
        ]}
        providers={[{ name: "aws" }]}
      />,
    );
    expect(view).toMatchSnapshot();
  });
});
