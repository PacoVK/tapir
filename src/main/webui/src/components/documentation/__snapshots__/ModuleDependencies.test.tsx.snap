// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleDependencies /> spec renders the ModuleDependencies 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
      >
        Module Dependencies
      </h4>
      <p
        class="MuiTypography-root MuiTypography-body1 css-t7zk8r-MuiTypography-root"
      >
         
        Dependencies are external modules that this module references. A module is considered external if it isn't within the same repository.
      </p>
      <div
        class="MuiStack-root css-uurwbu-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            <strong>
              terraform-aws-modules/security-group/aws
            </strong>
          </p>
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm css-g3yzuj-MuiContainer-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            >
              Version:
               
              <em>
                latest
              </em>
            </p>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-88ji4t-MuiDivider-root"
      />
      <h4
        class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
      >
        Provider Dependencies
      </h4>
      <p
        class="MuiTypography-root MuiTypography-body1 css-t7zk8r-MuiTypography-root"
      >
        Providers are Terraform plugins that will be automatically installed, if available on the Terraform Registry.
      </p>
      <div
        class="MuiStack-root css-gij6rf-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            <strong>
              aws
            </strong>
          </p>
          <div
            class="MuiContainer-root MuiContainer-maxWidthSm css-g3yzuj-MuiContainer-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
            >
              Version:
               
              <em>
                latest
              </em>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
    >
      Module Dependencies
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1 css-t7zk8r-MuiTypography-root"
    >
       
      Dependencies are external modules that this module references. A module is considered external if it isn't within the same repository.
    </p>
    <div
      class="MuiStack-root css-uurwbu-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          <strong>
            terraform-aws-modules/security-group/aws
          </strong>
        </p>
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-g3yzuj-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            Version:
             
            <em>
              latest
            </em>
          </p>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-88ji4t-MuiDivider-root"
    />
    <h4
      class="MuiTypography-root MuiTypography-h4 css-f84ov1-MuiTypography-root"
    >
      Provider Dependencies
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1 css-t7zk8r-MuiTypography-root"
    >
      Providers are Terraform plugins that will be automatically installed, if available on the Terraform Registry.
    </p>
    <div
      class="MuiStack-root css-gij6rf-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          <strong>
            aws
          </strong>
        </p>
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-g3yzuj-MuiContainer-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
          >
            Version:
             
            <em>
              latest
            </em>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
